imcsdk/apis/admin/ip.py

```diff
@@ -0,0 +1,328 @@
+# Copyright 2016 Cisco Systems, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+This module implements the APIs for IP Blocking and IP Filtering
+"""
+import logging
+from imcsdk.mometa.ip.IpBlocking import IpBlocking
+from imcsdk.mometa.ip.IpFiltering import IpFiltering, IpFilteringConsts
+from imcsdk.apis.utils import _get_mo, _is_valid_arg, _is_invalid_value
+from imcsdk.imccoreutils import get_server_dn, IMC_PLATFORM
+from imcsdk.imcexception import ImcOperationError
+
+log = logging.getLogger('imc')
+
+
+def _get_mgmt_if_dn(handle, id=1):
+    from imcsdk.mometa.mgmt.MgmtIf import MgmtIf
+
+    if handle.platform == IMC_PLATFORM.TYPE_CLASSIC:
+        parent_dn = get_server_dn(handle) + '/mgmt'
+    elif handle.platform == IMC_PLATFORM.TYPE_MODULAR:
+        parent_dn = 'sys/chassis-1'
+
+    mo = MgmtIf(parent_mo_or_dn=parent_dn)
+    return mo.dn
+
+
+def ip_blocking_enable(handle, fail_count=5, fail_window=60, penalty_time=300,
+                       **kwargs):
+    """
+    Enables IP Blocking
+
+    Args:
+        handle (ImcHandle)
+        fail_count (str): Number of times a user can attempt to log in
+                          unsuccessfully, before the system locks that user out
+                          Range [3-10] attempts
+        fail_window (str): Length of time, in seconds, in which the
+                           unsuccessful login attempts must occur in order
+                           for the user to be locked out.
+                           Range [60-120] seconds
+        penalty_time (str): The number of seconds the user remains locked out.
+                            Range [300-900] seconds
+        kwargs: Key-Value paired arguments for future use
+
+    Returns:
+        IpBlocking object
+
+    Examples:
+        ip_blocking_enable(handle, fail_count='6',
+                           fail_window='120', penalty_time='800')
+    """
```

Comment:
@waghswapnil this API is slightly confusing when it comes to param types.. 

the defaults specified are ints..
```
def ip_blocking_enable(handle, fail_count=5, fail_window=60, penalty_time=300, **kwargs):
```

the docstring says `str`
```
       fail_count (str): Number of times a user can attempt to log in
 +                          unsuccessfully, before the system locks that user out
 +                          Range [3-10] attempts
 +        fail_window (str): Length of time, in seconds, in which the
 +                           unsuccessful login attempts must occur in order
 +                           for the user to be locked out.
 +                           Range [60-120] seconds
 +        penalty_time (str): The number of seconds the user remains locked out.
 +                            Range [300-900] seconds
```

the example specifies them as `str`
```
+        ip_blocking_enable(handle, fail_count='6',
+                           fail_window='120', penalty_time='800')
```
when forming args internally we don't do `str()`.  We should clean this up for this and other APIs(if so).

The exists function also assumes `str`.